test_data/absolute_mayhem_fixed_output.txt

```diff
@@ -0,0 +1,246 @@
+║   │   │   │  │  │  │   │   │ ▓ │   │   │   │   │   │   3   ║
+║   │   │   │  │  │  │   │   │ 4 │   │   │   │   │   │       ║
+║   │   │   │  └──┘6 │   │   │ ▓ │   │   │   │   │ 5 │       ║
+╚═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══════╝
+ ╔═══╤═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══════╗ 
+┌──────────────────────────────────────────────────────────────┐
+│               The provided input is not valid!               │
+│                      Offending input: Oo                     │
+└──────────────────────────────────────────────────────────────┘
+ ╚═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══════╝ 
+ ╔═══╤═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══════╗ 
+┌──────────────────────────────────────────────────────────────┐
+│               The provided input is not valid!               │
+│                     Offending input: OoO                     │
+└──────────────────────────────────────────────────────────────┘
+ ╚═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══════╝ 
+ ╔═══╤═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══════╗ 
+┌──────────────────────────────────────────────────────────────┐
+│               The provided input is not valid!               │
+│                    Offending input: OoO...                   │
+└──────────────────────────────────────────────────────────────┘
+ ╚═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══════╝ 
+╔═══╤═══╤═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══════╗
+║   │   │   │   │  ┌──┐1 │   │ ▓ │   │   │   │   │   │   2   ║
+║   │   │   │   │  │  │  │   │ ▓ │   │   │   │   │   │   3   ║
+║   │   │   │   │  │  │  │   │ 4 │   │   │   │   │   │       ║
+║   │   │   │   │  └──┘6 │   │ ▓ │   │   │   │   │ 5 │       ║
+╚═══╧═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══════╝
+╔═══╤═══╤═══╤═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══╤═══╤═══════╗
+║   │   │   │   │   │  ┌──┐1 │ ▓ │   │   │   │   │   │   2   ║
+║   │   │   │   │   │  │  │  │ ▓ │   │   │   │   │   │   3   ║
+║   │   │   │   │   │  │  │  │ 4 │   │   │   │   │   │       ║
+║   │   │   │   │   │  └──┘6 │ ▓ │   │   │   │   │ 5 │       ║
+╚═══╧═══╧═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══╧═══╧═══════╝
+╔═══╤═══╤═══╤═══╤═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══╤═══════╗
+║   │   │   │   │   │ ▓ │  ┌──┐1 │   │   │   │   │   │   2   ║
+║   │   │   │   │   │ ▓ │  │  │  │   │   │   │   │   │   3   ║
+║   │   │   │   │   │ ▓ │  │4 │  │   │   │   │   │   │       ║
+║   │   │   │   │   │ ▓ │  └──┘6 │   │   │   │   │ 5 │       ║
+╚═══╧═══╧═══╧═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══╧═══════╝
+╔═══╤═══╤═══╤═══╤═══╤═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══════╗
+║   │   │   │   │   │ ▓ │   │  ┌──┐1 │   │   │   │   │   2   ║
+║   │   │   │   │   │ ▓ │   │  │  │  │   │   │   │   │   3   ║
+║   │   │   │   │   │ ▓ │   │  │4 │  │   │   │   │   │       ║
+║   │   │   │   │   │ ▓ │   │  └──┘6 │   │   │   │ 5 │       ║
+╚═══╧═══╧═══╧═══╧═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══════╝
```

Comment:
This output is wrong. When the boat moves on the central tile (where ▓ are), the ▓ characters must not be visible as the boat covers them. Lines 187-192 should look like:
```
╔═══╤═══╤═══╤═══╤═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══╤═══════╗
║   │   │   │   │   │   │  ┌──┐1 │   │   │   │   │   │   2   ║
║   │   │   │   │   │   │  │  │  │   │   │   │   │   │   3   ║
║   │   │   │   │   │   │  │4 │  │   │   │   │   │   │       ║
║   │   │   │   │   │   │  └──┘6 │   │   │   │   │ 5 │       ║
╚═══╧═══╧═══╧═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══╧═══════╝
```

Lines 193-198 must also be updated accordingly:

```
╔═══╤═══╤═══╤═══╤═══╤═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══════╗
║   │   │   │   │   │   │ ▓ │  ┌──┐1 │   │   │   │   │   2   ║
║   │   │   │   │   │   │ ▓ │  │  │  │   │   │   │   │   3   ║
║   │   │   │   │   │   │ ▓ │  │4 │  │   │   │   │   │       ║
║   │   │   │   │   │   │ ▓ │  └──┘6 │   │   │   │ 5 │       ║
╚═══╧═══╧═══╧═══╧═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══════╝
```

Reply:
one question regarding this case. I think in your output the initial position is no longer the central position. Because i would consider it central if there are 5 river tiles behind and 5 river tiles in front of the initial position (where ▓ are).

So instead of this:
```
╔═══╤═══╤═══╤═══╤═══╤═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══════╗
║   │   │   │   │   │   │ ▓ │  ┌──┐1 │   │   │   │   │   2   ║
║   │   │   │   │   │   │ ▓ │  │  │  │   │   │   │   │   3   ║
║   │   │   │   │   │   │ ▓ │  │4 │  │   │   │   │   │       ║
║   │   │   │   │   │   │ ▓ │  └──┘6 │   │   │   │ 5 │       ║
╚═══╧═══╧═══╧═══╧═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══════╝
```
this one is central
```
╔═══╤═══╤═══╤═══╤═══╤═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══════╗
║   │   │   │   │   │ ▓ │   │  ┌──┐1 │   │   │   │   │   2   ║
║   │   │   │   │   │ ▓ │   │  │  │  │   │   │   │   │   3   ║
║   │   │   │   │   │ ▓ │   │  │4 │  │   │   │   │   │       ║
║   │   │   │   │   │ ▓ │   │  └──┘6 │   │   │   │ 5 │       ║
╚═══╧═══╧═══╧═══╧═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══════╝
```

Reply:
@alessiogambi could you please clarify this?

Reply:
I do not understand your comment. The central tile is always in the same position. The boat moves on the tiles as fishes do. The tile on which the boat currently is placed must be bigger than the others to host the boat.

The entire board consists of 13 tiles, numbered from 0 to 12. The central tile is in position 6, the open sea is in position 12. At the beginning of the game, the boat is on tile 0.

```
TILE:  0     1   2   3   4   5   6   7   8   9   10  11   12
  ╔════════╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══════╗
  ║  ┌──┐1 │   │   │   │   │   │ 2 │   │   │   │   │   │       ║
  ║  │  │  │   │   │   │   │   │ 3 │   │   │   │   │   │       ║
  ║  │  │  │   │   │   │   │   │ 4 │   │   │   │   │   │       ║
  ║  └──┘6 │   │   │   │   │   │ 5 │   │   │   │   │   │       ║
  ╚════════╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══════╝
```
When the boat moves, it simply goes to the next tile. But the tiles remain always in their original positions.
For example, moving the boat two times we obtain:

```
   0    1      2     3   4   5   6   7   8   9   10  11   12
  ╔═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══╤═══════╗
  ║   │   │  ┌──┐1 │   │   │   │ 2 │   │   │   │   │   │       ║
  ║   │   │  │  │  │   │   │   │ 3 │   │   │   │   │   │       ║
  ║   │   │  │  │  │   │   │   │ 4 │   │   │   │   │   │       ║
  ║   │   │  └──┘6 │   │   │   │ 5 │   │   │   │   │   │       ║
  ╚═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══╧═══════╝
```

If the fishes move by one or two tiles (just to see the shaded chars) and the boat moves four times, it will reach tile 6 and the shaded chars cannot be seen:

```
   0    1   2   3   4   5     6      7   8   9   10  11   12
  ╔═══╤═══╤═══╤═══╤═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══╤═══════╗
  ║   │   │   │   │   │   │  ┌──┐1 │ 2 │   │   │   │   │       ║
  ║   │   │   │   │   │   │  │  │  │ 3 │   │   │   │   │       ║
  ║   │   │   │   │   │   │  │  │  │   │ 4 │   │   │   │       ║
  ║   │   │   │   │   │   │  └──┘6 │   │ 5 │   │   │   │       ║
  ╚═══╧═══╧═══╧═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══╧═══════╝
```

When the boat moves again by one tile, it moves away from the central tile and the shaded chars become visible again:

```
   0    1   2   3   4   5   6      7     8   9   10  11   12
  ╔═══╤═══╤═══╤═══╤═══╤═══╤═══╤════════╤═══╤═══╤═══╤═══╤═══════╗
  ║   │   │   │   │   │   │ ▓ │  ┌──┐1 │   │   │   │   │       ║
  ║   │   │   │   │   │   │ ▓ │  │23│  │   │   │   │   │       ║
  ║   │   │   │   │   │   │ ▓ │  │  │  │ 4 │   │   │   │       ║
  ║   │   │   │   │   │   │ ▓ │  └──┘6 │ 5 │   │   │   │       ║
  ╚═══╧═══╧═══╧═══╧═══╧═══╧═══╧════════╧═══╧═══╧═══╧═══╧═══════╝
```